sentry_streams/sentry_streams/pipeline/batch.py

```diff
@@ -0,0 +1,26 @@
+from typing import Self
+
+from sentry_streams.pipeline.function_template import Accumulator, InputType
+
+
+class BatchBuilder(Accumulator[InputType, list[InputType]]):
```

Comment:
List is mainly to illustrate, this should probably be a broader type which signifies a Collection